src/game/AI/ScriptDevAI/scripts/kalimdor/desolace.cpp

```diff
@@ -636,6 +471,74 @@ struct npc_cork_gizeltonAI : public ScriptedAI
             if (pPlayer->GetQuestStatus(QUEST_GIZELTON_CARAVAN) == QUEST_STATUS_INCOMPLETE)
                 pPlayer->FailQuest(QUEST_GIZELTON_CARAVAN);
         }
+
+        m_creature->GetCreatureGroup()->Despawn();
+    }
+
+    void UpdateEscortAI(const uint32 diff) override
+    {
+        npc_escortAI::UpdateEscortAI(diff);
+
+        if (ScriptDone)
+            return;
+
+        if (!StatusFailed)
+            return;
+
+        FailTimer -= diff;
+        if (FailTimer > 0 && !m_creature->GetCreatureGroup()->IsOutOfCombat())
+            return;
+
+        MakeQuestFailAndClean();
+        ScriptDone = true;
+    }
+
+    void WaypointReached(uint32 uiPointId) override
+    {
+        if (!HasEscortState(STATE_ESCORT_ESCORTING))
+            return;
+
+        Player* pPlayer = GetPlayerForEscort();
+
+        switch (uiPointId)
+        {
+        case 19:
+            if (pPlayer->GetQuestStatus(QUEST_BODYGUARD_TO_HIRE) == QUEST_STATUS_INCOMPLETE)
```

Comment:
These checks seem to be not good enough, as i got him to start glitching when i completed the first quest (north) and then done the second one (south)

![grafik](https://user-images.githubusercontent.com/19734826/194618495-7b682d78-4f2e-421b-ad3c-a01ca62549f6.png)

19:36:50 AIName:  -  (struct npc_cork_gizeltonAI) ScriptName: npc_cork_gizelton
19:36:50 EscortAI started for Player Testwarrior (Counter: 10) 
19:36:50 EscortFlags: Escorting
19:36:50 WaypointMovement: Last Reached WP: 32 (Loaded path 19019 from invalid origin)